tiledb/sm/filesystem/s3.h

```diff
@@ -481,21 +481,51 @@ class S3 {
       const URI& uri, const void* buffer, uint64_t length, bool last_part);
 
   /**
-   * Issues a multipart upload request.
+  * This struct wraps the context state of a pending multipart upload request.
+  */
+  struct MakeUploadPartCtx {
```

Comment:
Can you move this type definition to just before the private attributes? Sometimes we also add a `PRIVATE DATATYPES` block comment but it's not necessary.